### PR TITLE
chore(linter): enable errcheck and ineffassign linters

### PR DESCRIPTION
**Pull request by mangalaman93**

### CODE REVIEW

**Review by skrdgraph — APPROVED** [review]

**Comment by skrdgraph** [comment]
On the CI run .. this seems to have no issues.

I am checking locally to make sure this doesn't break after merge (like last time).

Screen Shot 2023-02-03 at 05 13 12

**Comment by skrdgraph** [comment]
We should not merge this, there is something wrong with this when run locally: 

```
Sudhishs-MacBook-Pro-Work:dgraph sudhishkr$ golangci-lint run


xidmap/trie.go:76:15: Error return value of `t.buf.Release` is not checked (errcheck)
        t.buf.Release()
                     ^
dgraph/cmd/live/run.go:191:19: Error return value of `txn.Discard` is not checked (errcheck)
        defer txn.Discard(ctx)
                         ^
dgraph/cmd/live/run.go:667:19: Error return value of `txn.Discard` is not checked (errcheck)
        defer txn.Discard(ctx)
                         ^
worker/backup_ee.go:327:17: Error return value of `th.buf.Release` is not checked (errcheck)
                th.buf.Release()
                              ^
worker/predicate_move.go:169:19: Error return value of `buf.SliceIterate` is not checked (errcheck)
                buf.SliceIterate(func(_ []byte) error {
                                ^
worker/predicate_move.go:289:20: Error return value of `buf.Release` is not checked (errcheck)
                defer buf.Release()
                                 ^
worker/restore_map.go:166:19: Error return value of `buf.Release` is not checked (errcheck)
        defer buf.Release()
                         ^
worker/restore_map.go:180:18: Error return value of `buf.SliceIterate` is not checked (errcheck)
        buf.SliceIterate(func(slice []byte) error {
                        ^
worker/restore_map.go:281:19: Error return value of `buf.Release` is not checked (errcheck)
        defer buf.Release()
                         ^
worker/restore_map.go:437:25: Error return value of `req.lbuf.Release` is not checked (errcheck)
                defer req.lbuf.Release()
                                      ^
worker/restore_reduce.go:255:17: Error return value of `cbuf.Release` is not checked (errcheck)
                                cbuf.Release()
                                            ^
worker/restore_reduce.go:270:15: Error return value of `cbuf.Release` is not checked (errcheck)
                cbuf.Release()
                            ^
worker/restore_reduce.go:284:16: Error return value of `kvBuf.Release` is not checked (errcheck)
                kvBuf.Release()
                             ^
worker/restore_reduce.go:311:15: Error return value of `cbuf.Release` is not checked (errcheck)
                cbuf.Release()
                            ^
dgraph/cmd/bulk/count_index.go:142:22: Error return value of `outBuf.Release` is not checked (errcheck)
        defer outBuf.Release()
                            ^
dgraph/cmd/bulk/count_index.go:166:18: Error return value of `buf.SliceIterate` is not checked (errcheck)
        buf.SliceIterate(func(slice []byte) error {
                        ^
dgraph/cmd/bulk/count_index.go:184:21: Error return value of `c.countBuf.Release` is not checked (errcheck)
                c.countBuf.Release()
                                  ^
dgraph/cmd/bulk/mapper.go:186:19: Error return value of `cbuf.SliceIterate` is not checked (errcheck)
        cbuf.SliceIterate(func(slice []byte) error {
                         ^
dgraph/cmd/bulk/mapper.go:284:19: Error return value of `sh.cbuf.Release` is not checked (errcheck)
                        sh.cbuf.Release()
                                       ^
dgraph/cmd/bulk/reduce.go:323:29: Error return value of `req.countBuf.Release` is not checked (errcheck)
                defer req.countBuf.Release()
                                          ^
dgraph/cmd/bulk/reduce.go:332:28: Error return value of `req.countBuf.SliceIterate` is not checked (errcheck)
                req.countBuf.SliceIterate(func(slice []byte) error {
                                         ^
dgraph/cmd/bulk/reduce.go:362:21: Error return value of `ci.writer.Write` is not checked (errcheck)
                                        ci.writer.Write(buf)
                                                       ^
dgraph/cmd/bulk/reduce.go:369:17: Error return value of `kvBuf.Release` is not checked (errcheck)
                        kvBuf.Release()
                                     ^
dgraph/cmd/bulk/reduce.go:423:19: Error return value of `cbuf.SliceIterate` is not checked (errcheck)
        cbuf.SliceIterate(func(slice []byte) error {
                         ^
dgraph/cmd/bulk/reduce.go:713:16: Error return value of `kvBuf.Release` is not checked (errcheck)
                kvBuf.Release()
                             ^
edgraph/server.go:1376:17: Error return value of `grpc.SendHeader` is not checked (errcheck)
        grpc.SendHeader(ctx, md)
                       ^
query/outputnode.go:961:6: ineffectual assignment to slice (ineffassign)
                                        slice = cur
                                        ^
query/outputnode.go:972:4: ineffectual assignment to slice (ineffassign)
                        slice = nil
                        ^
dgraph/cmd/bulk/reduce.go:604:3: ineffectual assignment to slice (ineffassign)
                slice, next := []byte{}, start
                ^
dgraph/cmd/live/run.go:614:6: ineffectual assignment to tlsConfig (ineffassign)
        var tlsConfig *tls.Config = nil
            ^
```

**Comment by skrdgraph** [comment]
convert to draft

**Comment by skrdgraph** [comment]
converting this to draft* (its still a WIP)

File: graphql/admin/http.go [hunk 1]
```diff
@@ -281,11 +281,13 @@ func (gh *graphqlHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 }
 
 func (gh *graphqlHandler) isValid(namespace uint64) error {
+	if gh == nil {
```
Comment (mangalaman93) [inline graphql/admin/http.go:284]: same here

**Comment by skrdgraph** [comment]
Just validated a local run - this looks good. Looks like the `golangci` ran fine. This should hold.

**Review by skrdgraph — APPROVED** [review]
👍

File: contrib/integration/acctupsert/main.go [hunk 2]
```diff
@@ -139,7 +141,12 @@ func tryUpsert(c *dgo.Dgraph, acc account) error {
 	ctx := context.Background()
 
 	txn := c.NewTxn()
-	defer func() { _ = txn.Discard(ctx) }()
+	defer func() {
+		if err := txn.Discard(ctx); err != nil {
+			glog.Warningf("error in discarding txn: %v", err)
```
Comment (matthewmcneely) [inline contrib/integration/acctupsert/main.go:146]: Nit: Log entries should be capitalized

File: dgraph/cmd/bulk/count_index.go [hunk 3]
```diff
@@ -111,7 +113,10 @@ func (c *countIndexer) addCountEntry(ce countEntry) {
 func (c *countIndexer) writeIndex(buf *z.Buffer) {
 	defer func() {
 		c.wg.Done()
-		buf.Release()
+		if err := buf.Release(); err != nil {
+			glog.Warningf("error in releasing buffer: %v", err)
```
Comment (matthewmcneely) [inline dgraph/cmd/bulk/count_index.go:117]: Nit: First word of log output should be capitalized

File: dgraph/cmd/bulk/count_index.go [hunk 4]
```diff
@@ -139,7 +144,11 @@ func (c *countIndexer) writeIndex(buf *z.Buffer) {
 	encoder := codec.Encoder{BlockSize: 256, Alloc: alloc}
 
 	outBuf := z.NewBuffer(5<<20, "CountIndexer.Buffer.WriteIndex")
-	defer outBuf.Release()
+	defer func() {
+		if err := outBuf.Release(); err != nil {
+			glog.Warningf("error in releasing buffer: %v", err)
```
Comment (matthewmcneely) [inline dgraph/cmd/bulk/count_index.go:149]: Nit: First word of log output should be capitalized

File: dgraph/cmd/bulk/count_index.go [hunk 5]
```diff
@@ -163,15 +172,18 @@ func (c *countIndexer) writeIndex(buf *z.Buffer) {
...
 		ce := countEntry(slice)
 		if !bytes.Equal(lastCe.Key(), ce.Key()) {
 			encode()
 		}
 		encoder.Add(ce.Uid())
 		lastCe = ce
 		return nil
-	})
+	}); err != nil {
+		glog.Errorf("error while counting in buf: %v\n", err)
```
Comment (matthewmcneely) [inline dgraph/cmd/bulk/count_index.go:184]: Nit: First word of log output should be capitalized

File: dgraph/cmd/bulk/count_index.go [hunk 6]
```diff
@@ -181,7 +193,9 @@ func (c *countIndexer) wait() {
 		c.wg.Add(1)
 		go c.writeIndex(c.countBuf)
 	} else {
-		c.countBuf.Release()
+		if err := c.countBuf.Release(); err != nil {
+			glog.Warningf("error in releasing buffer: %v", err)
```
Comment (matthewmcneely) [inline dgraph/cmd/bulk/count_index.go:197]: Nit: First word of log output should be capitalized

File: dgraph/cmd/bulk/mapper.go [hunk 7]
```diff
@@ -155,7 +156,9 @@ func (m *mapper) openOutputFile(shardIdx int) (*os.File, error) {
 func (m *mapper) writeMapEntriesToFile(cbuf *z.Buffer, shardIdx int) {
 	defer func() {
 		m.shards[shardIdx].mu.Unlock() // Locked by caller.
-		cbuf.Release()
+		if err := cbuf.Release(); err != nil {
+			glog.Warningf("error in releasing buffer: %v", err)
```
Comment (matthewmcneely) [inline dgraph/cmd/bulk/mapper.go:160]: Nit: First word of log output should be capitalized

File: dgraph/cmd/bulk/mapper.go [hunk 8]
```diff
@@ -281,7 +287,9 @@ func (m *mapper) run(inputFormat chunker.InputFormat) {
 			sh.mu.Lock() // One write at a time.
 			m.writeMapEntriesToFile(sh.cbuf, i)
 		} else {
-			sh.cbuf.Release()
+			if err := sh.cbuf.Release(); err != nil {
+				glog.Warningf("error in releasing buffer: %v", err)
```
Comment (matthewmcneely) [inline dgraph/cmd/bulk/mapper.go:291]: Nit: First word of log output should be capitalized

File: dgraph/cmd/bulk/reduce.go [hunk 9]
```diff
@@ -320,7 +320,11 @@ func (r *reducer) startWriting(ci *countIndexer, writerCh chan *encodeRequest, c
 	go r.writeTmpSplits(ci, tmpWg)
 
 	count := func(req *encodeRequest) {
-		defer req.countBuf.Release()
+		defer func() {
+			if err := req.countBuf.Release(); err != nil {
+				glog.Warningf("error in releasing buffer: %v", err)
```
Comment (matthewmcneely) [inline dgraph/cmd/bulk/reduce.go:325]: Nit: First word of log output should be capitalized

File: dgraph/cmd/bulk/reduce.go [hunk 10]
```diff
@@ -329,11 +333,14 @@ func (r *reducer) startWriting(ci *countIndexer, writerCh chan *encodeRequest, c
...
 		ci.countBuf.Grow(sz)
 
-		req.countBuf.SliceIterate(func(slice []byte) error {
+		if err := req.countBuf.SliceIterate(func(slice []byte) error {
 			ce := countEntry(slice)
 			ci.addCountEntry(ce)
 			return nil
-		})
+		}); err != nil {
+			glog.Errorf("error while iterating over buf: %v", err)
```
Comment (matthewmcneely) [inline dgraph/cmd/bulk/reduce.go:341]: Nit: First word of log output should be capitalized

File: graphql/admin/namespace.go [hunk 11]
```diff
@@ -67,7 +67,7 @@ func resolveDeleteNamespace(ctx context.Context, m schema.Mutation) (*resolve.Re
 	}
 	// No one can delete the galaxy(default) namespace.
 	if uint64(req.NamespaceId) == x.GalaxyNamespace {
-		return resolve.EmptyResult(m, errors.New("Cannot delete default namespace.")), false
+		return resolve.EmptyResult(m, errors.New("Cannot delete default namespace")), false
```
Comment (matthewmcneely) [inline graphql/admin/namespace.go:70]: Error messages should not be capitalized

**Review by harshil-goel — APPROVED** [review]

**Review by matthewmcneely — CHANGES_REQUESTED** [review]
Nit: There are dozens of places in this PR where the first word of log output is not capitalized. The Go conventions we follow require the opposite. I realize this isn't super-consistent across the code-base, but... (note: I gave up commenting on each occurrence after 9 or so)

**Comment by mangalaman93** [comment]
@matthewmcneely I think it is the opposite. Everything should start from small letters, nothing should be capitalized.

**Comment by matthewmcneely** [comment]
@mangalaman93 Well it's definitely inconsistent throughout. But IIRC, the convention is that the first word in output to logs *should* be capitalized. Error messages should NEVER be capitalized or punctutated, because they are likely to be chained together.

I guess we just need to declare which way is "the way" for Dgraph.

**Pull request merged**